libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1311,23 +1318,32 @@ string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
 					default {
 						sstore(array, add(data, 2))
 						let slot, offset := <indexAccess>(array, oldLen)
-						<storeValue>(slot, offset, value)
+						<storeValue>(slot, offset, <values>)
 					}
 				<!isByteArray>
 					let oldLen := sload(array)
 					if iszero(lt(oldLen, <maxArrayLength>)) { <panic>() }
 					sstore(array, add(oldLen, 1))
 					let slot, offset := <indexAccess>(array, oldLen)
-					<storeValue>(slot, offset, value)
+					<storeValue>(slot, offset, <values>)
 				</isByteArray>
 			})")
 			("functionName", functionName)
+			(
+				"values",
+				_fromType && _fromType->sizeOnStack() > 1 ?
```

Comment:
Actually what about multi-slot values that are given implicitly? What about just adding the following at the beginning:
`if (!fromType) fromType = _type.baseType()`?

Reply:
Fixed